packages/tools/client-debugger/client-debugger-view/src/components/DataObjectsView.tsx

```diff
@@ -18,10 +32,61 @@ export type DataObjectsViewProps = HasContainerId;
  * Dispatches data object rendering based on those provided view {@link DataObjectsViewProps.renderOptions}.
  */
 export function DataObjectsView(props: DataObjectsViewProps): React.ReactElement {
+	const { containerId } = props;
+
+	const messageRelay = useMessageRelay();
+
+	const [rootDataHandles, setRootDataHandles] = React.useState<
+		Record<string, RootHandleNode> | undefined
+	>();
+
+	React.useEffect(() => {
+		const inboundMessageHandlers: InboundHandlers = {
+			["ROOT_DATA_VISUALIZATIONS"]: (untypedMessage) => {
+				const message: RootDataVisualizationsMessage =
+					untypedMessage as RootDataVisualizationsMessage;
+
+				if (message.data.containerId === containerId) {
+					setRootDataHandles(message.data.visualizations);
+
+					return true;
+				} else {
+					return false;
+				}
+			},
+		};
+
+		function messageHandler(message: Partial<IDebuggerMessage>): void {
+			handleIncomingMessage(message, inboundMessageHandlers, {
+				context: loggingContext,
+			});
+		}
+
+		messageRelay.on("message", messageHandler);
+
+		messageRelay.postMessage({
```

Comment:
Can we leave a brief comment noting that we are requesting the initial data here? Makes it a bit easier to find this code within the effect while debugging.

Reply:
Comment added to tell that the POST is requested for DDS data inside the container.